reframe/core/logging.py

```diff
@@ -689,10 +688,22 @@ def emit(self, record):
             return
 
         try:
-            requests.post(
-                self._url, data=json_record,
-                headers=self._headers
-            )
+            while True:
+                response = requests.post(
+                    self._url, data=json_record,
+                    headers=self._headers
+                )
+                if response.status_code == 200:
```

Comment:
Maybe we should go for the more inclusive:

```suggestion
                if response.ok:
```